examples/vision/siamese.py

```diff
@@ -0,0 +1,397 @@
+"""
+Title: Siamese network with a contrastive loss
+Author: Mehdi
+Date created: 2021/05/06
+Last modified: 2021/05/06
+Description: Similarity learning using siamese network with contrastive loss
+"""
+
+"""
+## Introduction
+
+[Siamese Network](https://en.wikipedia.org/wiki/Siamese_neural_network)
+is any Neural Network which share weights between two or more sister networks,
+each producing embedding vector of its respective input and these embeddings
+are then passed through some 
+[distance heuristic](https://developers.google.com/machine-learning/clustering/similarity/measuring-similarity)
+to find the distance between them. This distance is later used to increase the
+contrast between embeddings of inputs of different classes and decrease it with
+that of similar class by employing some loss function, with the main objective
+of contrasting [vector spaces](https://en.wikipedia.org/wiki/Vector_space)
+from which these sample inputs were taken.
+"""
+
+"""
+## Setup
+"""
+
+import random
+import numpy as np
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+import matplotlib.pyplot as plt
+from sklearn.model_selection import train_test_split
+
+
+"""
+## Load the MNIST dataset
+"""
+
+(x_train, y_train), (x_test, y_test) = keras.datasets.mnist.load_data()
+
+# Change the data type to a floating point format
+x_train = x_train.astype("float32")
+x_test = x_test.astype("float32")
+
+
+"""
+Use train_test_split to split train data into `train` and `val`
+"""
+
+x_train, x_val, y_train, y_val = train_test_split(x_train, y_train, test_size=0.10)
```

Comment:
Please do this by hand, we don't need to add a dependency on sklearn just for `train_test_split`.

Reply:
sure